index.bs

```diff
@@ -80,271 +184,461 @@ Issue: Finish up this section.
-};
+[Exposed=Window, SecureContext]
+interface Screens {
+  readonly attribute FrozenArray<ScreenAdvanced> screens;
 
+  [SameObject]
+  readonly attribute ScreenAdvanced currentScreen;
 
-dictionary ScreenInfo {
-  // Shape matches https://drafts.csswg.org/cssom-view/#the-screen-interface
-  long availWidth;
-  long availHeight;
-  long width;
-  long height;
-  unsigned long colorDepth;
-  unsigned long pixelDepth;
-
-  // Shape roughly matches https://w3c.github.io/screen-orientation
-  OrientationType orientationType;  // Orientation type, e.g. "portrait-primary"
-  unsigned short orientationAngle;  // Orientation angle, e.g. 0
-
-  // Shape matches https://developer.mozilla.org/en-US/docs/Web/API/Screen
-  // Critical for understanding relative screen layouts for window placement.
-  // Distances from a multi-screen origin (e.g. primary screen top left) to the:
-  long left;       // Left edge of the screen area, e.g. 1920
-  long top;        // Top edge of the screen area, e.g. 0
-  long availLeft;  // Left edge of the available screen area, e.g. 1920
-  long availTop;   // Top edge of the available screen area, e.g. 0
-
-  // New properties critical for many multi-screen window placement use cases.
-  boolean isPrimary;
-  boolean internal;
-  float scaleFactor;     // Ratio between physical pixels and device
-                         // independent pixels for this screen, e.g. 2
-                         // Useful for placing windows on screens with optimal
-                         // scaling and appearances for a given application.
-  DOMString id;          // A temporary, generated per-origin unique ID; resets
-                         // when cookies are deleted. Useful for persisting user
-                         // window placements preferences for certain screens.
-  boolean touchSupport;
+  attribute EventHandler onscreenschange;
+  attribute EventHandler oncurrentscreenchange;
 };
 </xmp>
 
 <div algorithm>
 
-The <dfn method for=Window>getScreens()</dfn> method steps are:
+The <dfn attribute for=Screens>screens</dfn> getter steps are:
 
-1. Let |promise| be [=/a new promise=].
+1. Let |screens| be a new [=/list=].
 
-1. Run the following steps [=/in parallel=]:
+1. For each |screen| of [=/connected screens=]:
 
-    1. Let |screens| be a new [=/list=].
+    1. Let |a| be the {{ScreenAdvanced}} object describing |screen|.
 
-    1. For each |screen| [=connected screen=].
+    1. [=list/Append=] |a| to |screens|.
 
-        1. Let |info| be a new {{ScreenInfo}} dictionary describing |screen|.
+1. Return the result of [=list/sorting=] |screens| in ascending order with the [=/screen ordering=] algorithm.
 
-        1. [=list/Append=] |info| to |screens|.
+</div>
 
-    1. [=/Resolve=] |promise| with |screens|.
 
+<div algorithm>
 
-1. Return |promise|.
+The <dfn>screen ordering</dfn> algorithm defines a [=/connected screen=] |a| as less than a [=/connected screen=] |b| if the following steps return true:
```

Comment:
I think sorting the list of screens is still TBD.  Do you think this is something we should do?

Reply:
I thought I saw code in Chromium that did this with an operator< and everything, but I'm not finding it now. Maybe it was chromium side.

Anyway... yes, we need to define it. Not needed for the OT, but someone will ask.